BALSAMIC/assets/scripts/filter_SVs.py

```diff
@@ -0,0 +1,350 @@
+    return max_af, final_ctg
+
+
+def filter_vcf(vcf_path: Path):
+    """
+    Adds soft filters based on variant presence in normal.
+
+    :param vcf_path: Path to VCF
+
+    Outputs to standard out.
+    """
+
+    vcf = vcfpy.Reader.from_path(vcf_path)
+
+    # Update VCF header
+    vcf.header.add_info_line(
+        vcfpy.OrderedDict(
+            [
+                ("ID", "AF_T_MAX"),
+                ("Type", "Float"),
+                (
+                    "Description",
+                    "Max AF in tumor, for rows with merged overlapping variants",
+                ),
+            ]
+        )
+    )
+
+    vcf.header.add_info_line(
+        vcfpy.OrderedDict(
+            [
+                ("ID", "AF_N_MAX"),
+                ("Type", "Float"),
+                (
+                    "Description",
+                    "Max AF in normal, for rows with merged overlapping variants",
+                ),
+            ]
+        )
+    )
+
+    vcf.header.add_filter_line(
+        vcfpy.OrderedDict(
+            [
+                ("ID", "normal_variant"),
+                ("Description", "AF_T_MAX == 0 and ctg_t == False"),
+            ]
+        )
+    )
+
+    vcf.header.add_filter_line(
+        vcfpy.OrderedDict(
+            [("ID", "high_normal_af"), ("Description", "AF_N_MAX > 0.25")]
+        )
+    )
+
+    vcf.header.add_filter_line(
+        vcfpy.OrderedDict(
+            [
+                ("ID", "high_normal_af_fraction"),
+                ("Description", "(AF_N_MAX / AF_T_MAX) > 0.25"),
+            ]
+        )
+    )
+
+    vcf.header.add_filter_line(
+        vcfpy.OrderedDict(
+            [
+                ("ID", "in_normal"),
+                ("Description", "ctg_n == True and AF_N_MAX == 0 and AF_T_MAX <= 0.25"),
+            ]
+        )
+    )
+
+    writer = vcfpy.Writer.from_path("/dev/stdout", vcf.header)
+
+    # Set soft filters for variants based on presence in the normal sample
+    for sv in vcf:
+        info = dict([*sv.INFO.items()])
+
+        # Set default values
+        ctg_t, ctg_n = False, False
+        af_t, af_n = 0, 0
+
+        if "TUMOR_PASS_SAMPLE" in info:
+            pass_sample = info["TUMOR_PASS_SAMPLE"]
+            pass_info = info["TUMOR_PASS_INFO"]
+            af_t, ctg_t = calc_af_get_ctg(pass_sample, pass_info)
+        if "NORMAL_PASS_SAMPLE" in info:
+            pass_sample = info["NORMAL_PASS_SAMPLE"]
+            pass_info = info["NORMAL_PASS_INFO"]
+            af_n, ctg_n = calc_af_get_ctg(pass_sample, pass_info)
+
+        # Set filter statuses
+        if af_t == 0 and not ctg_t:
+            sv.add_filter("normal_variant")
+        else:
+            # Regardless of CTG, set filter if AF_T / AF_N > 0.25
+            if af_t > 0:
+                if float(af_n / af_t) > 0.25:
+                    normal_af_status = "failed"
+                    sv.add_filter("high_normal_af_fraction")
+                else:
+                    normal_af_status = "pass"
```

Comment:
Maybe something like this would simplify this part?:
```suggestion
            # Calculate AF_T / AF_N ratio
            af_ratio = float(af_n / af_t) if af_t > 0 else 0
            if af_ratio > 0.25:
                sv.add_filter("high_normal_af_fraction")
```

Reply:
That works, and places the emphasis on the right area. I agree it looks better. Implemented!